integrations/jira/event.go

```diff
@@ -41,63 +41,76 @@ func NewHTTPHandler(l *zap.Logger, o sdkservices.OAuth, v sdkservices.Vars, d sd
 	return handler{logger: l, oauth: o, vars: v, dispatcher: d}
 }
 
-type event struct {
-	WebhookEvent      string `json:"webhookEvent"`
-	MatchedWebhookIDs []int  `json:"matchedWebhookIds"`
-	Timestamp         int64  `json:"timestamp"`
-}
-
 // handleEvent receives from Jira asynchronous events,
 // and dispatches them to one or more AutoKitteh connections.
 // Note 1: By default, AutoKitteh creates webhooks automatically,
 // subscribing to all events - see "webhooks.go" for more details.
 // TODO(ENG-965):
 // Note 2: Dynamic (i.e. auto-created) webhooks expire after 30 days.
 // This functions extends this deadline at the 20-day mark.
+// Note 3: The requests are sent by a service, so no need to respond
+// with user-friendly error web pages.
 func (h handler) handleEvent(w http.ResponseWriter, r *http.Request) {
 	l := h.logger.With(zap.String("urlPath", r.URL.Path))
 
 	// Verify the JWT in the event's "Authorization" header.
 	token := r.Header.Get(headerAuthorization)
 	if !verifyJWT(l, strings.TrimPrefix(token, "Bearer ")) {
-		// Attack or network loss, so no need for user-friendliness.
+		l.Warn("Incoming Jira event with bad header", zap.String(headerAuthorization, token))
 		http.Error(w, "Unauthorized", http.StatusUnauthorized)
 		return
 	}
 
 	// Check the "Content-Type" header.
 	contentType := r.Header.Get(headerContentType)
 	if !strings.HasPrefix(contentType, contentTypeJSON) {
-		// Probably an attack, so no need for user-friendliness.
+		l.Warn("Incoming Jira event with bad header", zap.String(headerContentType, contentType))
 		http.Error(w, "Bad Request", http.StatusBadRequest)
 		return
 	}
 
 	// Parse some of the metadata in the Jira event's JSON content.
 	body, err := io.ReadAll(r.Body)
 	if err != nil {
-		// Attack or network loss, so no need for user-friendliness.
+		l.Warn("Failed to read content of incoming Jira event", zap.Error(err))
 		http.Error(w, "Bad Request", http.StatusBadRequest)
 		return
 	}
 
-	var e event
-	if err := json.Unmarshal(body, &e); err != nil {
-		// Probably an attack, so no need for user-friendliness.
+	var jiraEvent map[string]interface{}
+	if err := json.Unmarshal(body, &jiraEvent); err != nil {
+		l.Warn("Failed to unmarshal JSON in incoming Jira event", zap.Error(err))
```

Comment:
error?

Reply:
and all above

Reply:
I'm using `Error` logs for things that AK messed up or suffers from, and `Warn` for issues that require attention but are not our fault/problem. In this case all these failures are because of external data (which should be reliable, but is beyond our control) and although they represent event loss and may need to be investigated and improved, it's not a misbehavior of the server.